### ISSUE TITLE
Data merging when plotting multiple charts.

### ISSUE THREAD

**Issue by jithinktom** [highcharts/highcharts-react#219]
Hi,

I have created a component called Chart, which renders the HighchartsReact component. I am rendering two Chart components on a page. In my Chart component, I am connecting to WebSocket and getting data dynamically. Each chart will listen to different data depending upon the props it receives. After getting data from Websocket, I add those data to the HighStock chart series, using the addPoint() method on the ref element. So ideally, each Chart component should be updated dynamically with its own data. The problem I am facing is, I am getting very weird shapes when plotting the two charts together. Also, in the console, I get error#15. They both work fine when I plot them individually. Can you help me with this?

```
import React from "react";
import Highcharts from "highcharts/highstock";
import HighchartsReact from "highcharts-react-official";
import HC_more from "highcharts/highcharts-more";
import worker from 'workerize-loader!./worker'; // eslint-disable-line import/no-webpack-loader-syntax

HC_more(Highcharts);

class Chart extends React.PureComponent {
    constructor(props) {
        super(props);
        this.chartComponent = React.createRef();

        this.state = {
            options: {
                time: {
                    useUTC: false
                },

                rangeSelector: {
                    buttons: [{
                        count: 1,
                        type: 'minute',
                        text: '1M'
                    }, {
                        type: 'all',
                        text: 'All'
                    }],
                    inputEnabled: false,
                    selected: 'all'
                },

                exporting: {
                    enabled: false
                },

                xAxis: {
                    type: "datetime"
                },
                navigator: {
                    adaptToUpdatedData: true
                },
                title: {
                    text: props.title
                },
                chart: {
                    events: {
                        load: function () {
                            //redraw chart manually on every 2 second
                            this.interval = setInterval(() => {
                                this.redraw()
                            }, 2000)
                        }
                    }
                },
                series: {
                    name: this.props.name,
                    data: [],
                    type: this.props.chartType
                }
            }
        };
    }

    componentDidMount() {
        //getting data from websocket. It is a web-worker
        let instance = worker()
        instance.expensive(Object.keys(this.props.initialData), this.props.topic);
        instance.onmessage = (m) => {
            if (m.data && m.data.type === "socketdata") {
                this.appendData(m.data.data)
            }
        };
    }

    componentWillUnmount() {
        clearInterval(this.interval);
    }

    appendData = (data) => {
        // function for adding points to chart
        let chart = this.chartComponent.current.chart;
        if (chart) {
            var series = chart.series;
            if (series)
                series[0].addPoint(data.data, false, series[0].data.length > 500 ? true : false);
        }
    }

    render() {
        return (
            <div className="chart-section">
                <HighchartsReact
                    constructorType={"stockChart"}
                    ref={this.chartComponent}
                    highcharts={Highcharts}
                    options={this.state.options}
                    immutable={true}
                />
            </div>
        );
    }
}

export default Chart;
```

![image](https://user-images.githubusercontent.com/10219327/82232831-c16e0a80-994c-11ea-9d9b-c6d6cbb4b876.png)

**Comment by ppotaczek** [comment]
HI @jithinktom,

Thank you for using our wrapper!

It looks like that you have unsorted data. Could you provide me with sample data that you received through WebSocket?

Best regards!

**Comment by jithinktom** [comment]
Hi @ppotaczek ,

Thanks for your reply. What was happening was, I was passing an array of data from my redux store to the chart at the time of initialization. After that, when the points were added to the chart, the data I passed inside was mutated. So basically, my redux store was mutated because of the reference. So what I did was, I cloned the data using "clone" library and then passed the data to chart. Now the problem is solved!
